### PR TITLE
add invalid check on build and search for hgraph

### PR DESCRIPTION
**Pull request by LHT129**

### CODE REVIEW

**Review by inabao — APPROVED** [review]
LGTM

File: src/index/hgraph_index.cpp [hunk 1]
```diff
@@ -95,6 +95,11 @@ tl::expected<std::vector<int64_t>, Error>
 HGraphIndex::add(const DatasetPtr& data) {
     std::vector<int64_t> failed_ids;
     try {
+        auto base_dim = data->GetDim();
+        CHECK_ARGUMENT(base_dim == dim_,
+                       fmt::format("base.dim({}) must be equal to index.dim({})", base_dim, dim_));
```
Comment (jiaweizone) [inline src/index/hgraph_index.cpp:100]: data.dim({}) must be equal to index.dim({}) ?

File: src/index/hgraph_index.cpp [hunk 2]
```diff
@@ -95,6 +95,11 @@ tl::expected<std::vector<int64_t>, Error>
...
+        CHECK_ARGUMENT(data->GetFloat32Vectors() != nullptr, "base.float_vector is nullptr");
```
Comment (jiaweizone) [inline src/index/hgraph_index.cpp:101]: data.float_vector is nullptr

Comment (LHT129) [reply src/index/hgraph_index.cpp:101]: actually base means "data of the database"

File: src/index/hgraph_index.cpp (hunk 1 above)
Comment (LHT129) [reply src/index/hgraph_index.cpp:100]: yep, 32dim index must use 32dim data to build

**Pull request merged**